### PR TITLE
Add popup menu to show network information

### PR DESCRIPTION
**Pull request by d-loose**
- add separate buttons to start/stop/delete an instance
- add popup menu button that can potentially replace the context menu :)

image
image
image

### CODE REVIEW

**Comment by jpnurmi** [comment]
Looking good so far! We'll eventually match this with the `lxc info <instance>` command that shows various information for an instance.

We could merge the service changes right away if you don't mind sending them as a separate PR. :)

**Comment by d-loose** [comment]
Ah yes, I can split that off :)

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -41,12 +41,15 @@
...
   "nameLabel": "Name",
+  "networkInformation": "Network information",
+  "networkInterface": "Interface",
   "nextTabCommand": "Next tab",
   "newTabCommand": "New tab",
   "newTabMenuItem": "New Tab",
   "okButton": "Ok",
   "openPaletteCommand": "Open command palette",
   "openPreferencesCommand": "Open preferences",
+  "openShell": "Open shell",
```
Comment (jpnurmi) [inline lib/l10n/app_en.arb:52]: can we call it "terminal"?

File: lib/home/instance_tile.dart [hunk 2]
```diff
@@ -17,21 +18,37 @@ class InstanceTile extends StatelessWidget {
...
+          List<PopupMenuEntry<Intent>> popupMenuBuilder(BuildContext context) =>
+              [
+                PopupMenuItem(
+                  value: SelectInstanceIntent(instance),
+                  child: Text(l10n.openShell),
+                ),
+                PopupMenuItem(
+                  value: ShowInstanceInfoIntent(instance),
+                  enabled: canStop,
+                  child: Text(l10n.networkInformation),
```
Comment (jpnurmi) [inline lib/home/instance_tile.dart:48]: could this be "Instance information"?

File: lib/home/instance_info_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        children: [
+          TableRow(
+              children: [
+            Text(l10n.networkInterface),
+            const Text('IPv4'),
+            const Text('IPv6')
+          ]
+                  .map((e) =>
+                      Padding(padding: const EdgeInsets.all(8), child: e))
+                  .toList()),
```
Comment (jpnurmi) [inline lib/home/instance_info_dialog.dart:31]: there's something wonky with the formatting here. would it help to add a comma after toList()?

**Review by jpnurmi — COMMENTED** [review]
there are some old network helper extensions in lxd_x.dart that might be useful

https://github.com/canonical/workshops/blob/main/packages/lxd_x/lib/lxd_x.dart#L96-L112

File: lib/l10n/app_en.arb (hunk 1 above)
Comment (d-loose) [reply lib/l10n/app_en.arb]: :+1:

File: lib/home/instance_tile.dart (hunk 2 above)
Comment (d-loose) [reply lib/home/instance_tile.dart]: :+1:

File: lib/home/instance_info_dialog.dart (hunk 3 above)
Comment (d-loose) [reply lib/home/instance_info_dialog.dart]: :+1:

**Review by jpnurmi — APPROVED** [review]
Thanks, LGTM :+1:

**Pull request merged**